organize/models.py

```diff
@@ -49,17 +49,29 @@ def create(self, **data_dict):
         )
 
         if previous_event:
-            event_date = data_dict["date"]
-            if date(event_date.year, event_date.month, event_date.day) - date(
-                previous_event.date.year, previous_event.date.month, previous_event.date.day
-            ) < timedelta(days=180):
-                raise ValidationError(
-                    {
-                        "date": _(
-                            "Your workshops should be at least 6 months apart. " "Please read our Organizer Manual."
-                        )
-                    }
-                )
+            try:
+                event_date = data_dict["date"]
+                if date(event_date.year, event_date.month, event_date.day) - date(
+                    previous_event.date.year, previous_event.date.month, previous_event.date.day
+                ) < timedelta(days=180):
+                    raise ValidationError(
+                        {
+                            "date": _(
+                                "Your workshops should be at least 6 months apart. " "Please read our Organizer Manual."
+                            )
+                        }
+                    )
+            except ValueError:
+                if date(event_date.year, event_date.month, event_date.day) - date(
+                    previous_event.date.year, previous_event.date.month, 1
```

Comment:
Is the `previous_event` day the only possible issue with the data that'll be provided here?

Reply:
Not sure but the error was resulting due to `previous_event` being an absolute date with just month and year. I guess maybe we could cater for both scenarios @marksweb ?

Reply:
Yeah, that gets a bit tricky though I bet. Maybe deploy this as is and we can keep an eye on errors should it need improvement.

Reply:
Noted.